Vizzuality/marxan-cloud
# setup for PostgreSQL+PostGIS container for API data layer

## PostgreSQL + PostGIS setup

### Overview

This adds a PostgreSQL container (Dockerfile + docker compose configuration), and the envvars needed to connect to a PostgreSQL instance/db from the API.

API setup for PostgreSQL will be in a separate PR.

### Designs

N/A

### Testing instructions

Define new envvars documented in README in the top-level .env file. Do note that some of the existing envvars will need renaming (I changed these to clarify what they are for, as in my experience envvar names tend to become an unwieldy mess if not kept under control)

`make start` should spin up the services including a PostgreSQL database. You will need to comment out the whole section of service definition for `app` in the docker compose config file because the initial config for a dummy app service needs to be updated for the current NextJS starter app/setup so docker compose will exit with an error if trying to build and run the app as currently defined in `docker-compose.yml`.

### Feature relevant tickets

https://www.pivotaltracker.com/story/show/176205161

---

## Checklist before submitting

      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
